packages/daemon/src/types.d.ts

```diff
@@ -453,7 +453,7 @@ export interface NameHub {
 }
 
 export interface EndoDirectory extends NameHub {
-  makeDirectory(petName: string): Promise<EndoDirectory>;
+  makeDirectory(...petNamePath: string[]): Promise<EndoDirectory>;
```

Comment:
This is fine, but I have misgivings about throwing away the possibility of cleanly taking an options bag later that would be addressed by taking `string | string[]` here instead. I wouldn’t mind if `lookup` were very special and don’t love that it _is_ special.

Reply:
Hm, we have actually accumulated a lot of these `...path` signature functions:
```ts
export interface NameHub {
  has(...petNamePath: string[]): Promise<boolean>;
  identify(...petNamePath: string[]): Promise<string | undefined>;
  locate(...petNamePath: string[]): Promise<string | undefined>;
  reverseLocate(locator: string): Promise<string[]>;
  followLocatorNameChanges(
    locator: string,
  ): AsyncGenerator<LocatorNameChange, undefined, undefined>;
  list(...petNamePath: string[]): Promise<Array<string>>;
  listIdentifiers(...petNamePath: string[]): Promise<Array<string>>;
  followNameChanges(
    ...petNamePath: string[]
  ): AsyncGenerator<PetStoreNameChange, undefined, undefined>;
  lookup(...petNamePath: string[]): Promise<unknown>;
  reverseLookup(value: unknown): Array<string>;
  write(petNamePath: string | string[], id: string): Promise<void>;
  remove(...petNamePath: string[]): Promise<void>;
  move(fromPetName: string[], toPetName: string[]): Promise<void>;
  copy(fromPetName: string[], toPetName: string[]): Promise<void>;
}

export interface EndoDirectory extends NameHub {
  makeDirectory(...petNamePath: string[]): Promise<EndoDirectory>;
}
```